docs/sources/reference/components/prometheus.exporter.snmp.md

```diff
@@ -187,13 +198,137 @@ prometheus.remote_write "demo" {
+prometheus.exporter.snmp "example" {
+    config_file = "snmp_modules.yml"
+
+    targets = [
+        {
+            "name"        = "network_switch_1",
+            "address"     = "192.168.1.2",
+            "module"      = "if_mib",
+            "walk_params" = "public",
+        },
+        {
+            "name"        = "network_router_2",
+            "address"     = "192.168.1.3",
+            "module"      = "mikrotik",
+            "walk_params" = "private",
+        },
+    ]
+
+    walk_param "private" {
+        retries = "2"
+    }
+
+    walk_param "public" {
+        retries = "2"
+    }
+}
+
+// Configure a prometheus.scrape component to collect SNMP metrics.
+prometheus.scrape "demo" {
+    targets    = prometheus.exporter.snmp.example.targets
+    forward_to = [ /* ... */ ]
+}
+```
+
+This example uses the [`local.file` component][file] to read targets from a YAML file and send them to the `prometheus.exporter.snmp` component:
+
+```alloy
+local.file "targets" {
+  filename = "targets.yml"
+}
+
+prometheus.exporter.snmp "example" {
+    config_file = "snmp_modules.yml"
+
+    targets = yaml_decode(local.file.targets.content)
+
+    walk_param "private" {
+        retries = "2"
+    }
+
+    walk_param "public" {
+        retries = "2"
+    }
+}
+
+// Configure a prometheus.scrape component to collect SNMP metrics.
+prometheus.scrape "demo" {
+    targets    = prometheus.exporter.snmp.example.targets
+    forward_to = [ /* ... */ ]
+}
+```
+
+The YAML file in this example looks like this:
+```
+- name: t1
+  address: localhost:161
+  module: default
+  auth: public_v2
+- name: t2
+  address: localhost:161
+  module: default
+  auth: public_v2
+```
+
+This example uses the [`discovery.file` component][disc] and the [`discovery.relabel` component][relabel] to send targets to the `prometheus.exporter.snmp` component:
+```
+discovery.file "example" {
+  files = ["targets.yml"]
+}
+
+discovery.relabel "example" {
+  targets = discovery.file.example.targets
+
+  rule {
+    source_labels = ["__address__"]
+    target_label  = "address"
+    action        = "replace"
+  }
+}
+
+prometheus.exporter.snmp "example" {
+  config_file = "snmp_modules.yml"
+  targets = discovery.relabel.example.output
+}
+
+// Configure a prometheus.scrape component to collect SNMP metrics.
+prometheus.scrape "demo" {
+    targets    = prometheus.exporter.snmp.example.targets
+    forward_to = [ /* ... */ ]
+}
+```
+
+The YAML file in this example looks like this:
+```
+- targets:
+  - localhost:161
+  labels:
+    name: t1
+    module: default
+    auth: public_v2
+- targets:
+  - localhost:161
+  labels:
+    name: t2
+    module: default
+    auth: public_v2
+```
+
 Replace the following:
 
 - `PROMETHEUS_REMOTE_WRITE_URL`: The URL of the Prometheus remote_write-compatible server to send metrics to.
 - `USERNAME`: The username to use for authentication to the remote_write API.
 - `PASSWORD`: The password to use for authentication to the remote_write API.
```

Comment:
```suggestion
Replace the following:

- _`<PROMETHEUS_REMOTE_WRITE_URL>`_: The URL of the Prometheus remote_write-compatible server to send metrics to.
- _`<USERNAME>`_: The username to use for authentication to the remote_write API.
- _`<PASSWORD>`_: The password to use for authentication to the remote_write API.
```

This text needs to moved up around line 200 -it applies to the generic example there, and is out of place down here at line 322.

In the generic example (lines 191-195) there also needs to be a small change ( I can't comment there)... wrap the replaceable args with `<` and `>`.

```
endpoint {
        url = <PROMETHEUS_REMOTE_WRITE_URL>
        basic_auth {
            username = LUSERNAME>
            password = <PASSWORD>
```